internal/porchutil/function.go

```diff
@@ -0,0 +1,155 @@
+package porchutil
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes"
+
+	// Enable the GCP Authentication plugin
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const expirationNanoSecond = 3
```

Comment:
that's very short amount of time :)

Reply:
The IETF says [current request timeout implementation select times between 30-120s](https://tools.ietf.org/id/draft-thomson-hybi-http-timeout-00.html#:~:text=Knowing%20the%20time%20available%20to,empirically%20determined%20to%20be%20safe.)

Maybe we can set that to 30s?

Reply:
this is for tab completion, correct? IMO 30s is too long.
let's try some value now (I'd recommend using the Duration type) and we'll adjust.

Reply:
I wonder what the experience is like in `porch` disconnected mode, does it fail fast ? (you might need two timeout, one for connectionTimeout, another one timing out on waiting from response).